### ISSUE TITLE
[A11y] copy-ansers-button in summary is buggy for keyboard users

### ISSUE THREAD

**Issue by wiffbi** [pretix/pretix#2227]
**Describe the bug**
When navigating the page/form with keyboard input only, the button to copy answers is buggy. Hitting enter to copy answers also triggers details/summary fold-in/out. Furthermore focus behaviour seems broken as summary as well as the button get a focus ring – maybe because both get triggered?

**Expected behavior**
When acitvating the copy-button, the summary does not get folded/unfolded. Also the focus ring should only be on summary or button.

**Comment by wiffbi** [comment]
The doubled focus line comes from :active-state. Problem is the `<button>` inside the `<summary>` – this is technically challenging as `<summary>` itself has the aria-role="button", so this results in a button inside a button. Therefore it is also not possible to stopPropagation() from the button to disable the summary trigger. 

TLDR; we need to place the copy-answer button somewhere else – at least in the HTML. Visually it could stay where it is, but it needs to be outside the `<summary>` element.

**Comment by wiffbi** [comment]
When keeping the `<button>` inside the `<summary>` visually, I need to place it before the whole `<details>`element, because otherwise it would be hidden. This makes the button appear in the wrong place regarding source order/structure. The only way to make this understandable for SR-users, is to add extensive description whereto the answers from above will be copied to.

An alternative might be to place the button inside the panel wo which it copies the answers to. Visually it could pick up the design from customer profiles as a kind of meta-data button/form-element. Like so:

![Bildschirmfoto 2021-09-29 um 13 43 09](https://user-images.githubusercontent.com/276509/135262629-f1396ac3-00d3-4b5e-843a-0af78d99261e.png)

Could even work floating on the right when a customer profile has been matched – although this needs experimentation how this behaves on small to middle-sized screens where floating right would look strange. It probably needs to align left with the second column of the grid. But this might be to close to the profile thingie and confuse users …

**Comment by wiffbi** [comment]
Floating the button to the left works surprisingly well on mobile. I will try this with real code and test it in different configurations.

![Bildschirmfoto 2021-10-04 um 15 50 22](https://user-images.githubusercontent.com/276509/135865369-a97b7be6-e27d-448b-ba35-39c98c11b4d9.png)

**Comment by wiffbi** [comment]
Hm, with mid-sized screens it’s a mixed bag. Floating on the same line as „Auto-fill with profile“ makes it a bit confusing.

![Bildschirmfoto 2021-10-04 um 16 17 13](https://user-images.githubusercontent.com/276509/135867742-43614571-cd98-4802-921c-6edff1618da6.png)

Not floating it, makes the margin on the left very big.

![Bildschirmfoto 2021-10-04 um 16 19 29](https://user-images.githubusercontent.com/276509/135868109-e36f28e7-cf4f-4145-8a26-c4450e13212d.png)

Original position for comparison:

![Bildschirmfoto 2021-10-04 um 16 23 17](https://user-images.githubusercontent.com/276509/135868823-cec31952-d9a2-4e0d-b3cf-09f3d67f47a6.png)

… which kind of breaks on smaller screens

![Bildschirmfoto 2021-10-04 um 16 27 59](https://user-images.githubusercontent.com/276509/135869484-cd0c8952-839f-4693-88c5-1cdcbad428f7.png)

Another alternative might be to place it right above the questions, but inside the white block belonging to the questions.

![Bildschirmfoto 2021-10-04 um 16 32 40](https://user-images.githubusercontent.com/276509/135870476-f698f4fa-61a3-4d2f-ae01-969e6f180b29.png)
![Bildschirmfoto 2021-10-04 um 16 33 15](https://user-images.githubusercontent.com/276509/135870499-888d957f-6fc7-4fc8-8094-07efe1e976ee.png)



Sidenote: the following translation might be better „Antworten von oben _übernehmen_“ (instead of „kopieren“).

**Comment by raphaelm** [comment]
Hmm, in all the screenshots where both profiles as well as copy-from-above are available, I think it will always get *pretty* confusing if we try to present them as separate concepts.

For example, if the "copy from above" button is *below* the auto-fill block like in your last screenshot, it becomes unclear what "from above"/"von oben" refers to – it could just as well be the suggested profile!

I think this means the copy-from-above button needs to be at the same "hierarchy level" as the "Auto-fill", leaving the floating approach.

Spontaneous thought: I'm thinking if we should maybe just drop the "copy **from** above" button on every ticket and add a "copy **to** all other tickets below" on the first ticket? It's less flexible, but might be more clear.

**Comment by wiffbi** [comment]
Yeah, the more I think about it, "copy/transfer? answers to all other tickets below" makes sense! It is indeed less flexible, but more clear. And IMHO usually the option I want anyways. So we place the button below the first set of answers. I will check how that looks especially with add-ons.
